src/coreclr/vm/pgo.cpp

```diff
@@ -988,22 +1175,28 @@ CORINFO_CLASS_HANDLE PgoManager::getLikelyClass(MethodDesc* pMD, unsigned ilSize
                 {
                     // Find maximum entry and return it
                     //
-                    unsigned maxIndex = 0;
                     unsigned maxCount = 0;
+                    unsigned maxKnownIndex = 0;
+                    unsigned maxKnownCount = 0;
 
                     for (unsigned m = 0; m < h.m_histogram.GetCount(); m++)
                     {
+                        if ((h.m_histogram[m].m_count > maxKnownCount) && !IsUnknownTypeHandle(h.m_histogram[m].m_mt))
+                        {
+                            maxKnownIndex = m;
+                            maxKnownCount = h.m_histogram[m].m_count;
+                        }
                         if (h.m_histogram[m].m_count > maxCount)
                         {
-                            maxIndex = m;
                             maxCount = h.m_histogram[m].m_count;
                         }
                     }
 
-                    if (maxCount > 0)
+                    UINT32 maxKnownLikelihood = (100 * maxKnownCount) / h.m_totalCount;
+                    if ((maxKnownCount > 0) && ((maxKnownCount == maxCount) || (maxKnownLikelihood > 33)))
```

Comment:
I don't think we should filter results here. The jit can decide if likelihood warrants testing things (currently we have different thresholds for virtual and interface calls, for instance).

Reply:
Ok. Will do.